browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchClient.java

```diff
@@ -112,73 +113,145 @@ public void start(
-     * @param activity the activity used to start browser switch
+     * @param browserSwitchOptions {@link BrowserSwitchOptions}
+     * @param activity the activity used to start browser switch. Must implement {@link BrowserSwitchListener}
      */
-    @SuppressWarnings("SameParameterValue")
-    public void start(int requestCode, Intent intent, FragmentActivity activity) {
+    public void start(BrowserSwitchOptions browserSwitchOptions, FragmentActivity activity) {
         if (activity instanceof BrowserSwitchListener) {
-            start(requestCode, intent, activity, (BrowserSwitchListener) activity);
+            start(browserSwitchOptions, activity, (BrowserSwitchListener) activity);
         } else {
             throw new IllegalArgumentException("Activity must implement BrowserSwitchListener.");
         }
     }
 
     /**
      * Open a browser or <a href="https://developer.chrome.com/multidevice/android/customtabs">Chrome Custom Tab</a>
-     * with the given intent from an Android activity.
+     * with a given set of {@link BrowserSwitchOptions} from an Android activity.
      *
-     * @param requestCode the request code used to differentiate requests from one another.
-     * @param intent the intent to use to initiate a browser switch
+     * @param browserSwitchOptions {@link BrowserSwitchOptions}
      * @param activity the activity used to start browser switch
      * @param listener the listener that will receive browser switch callbacks
      */
-    public void start(
-        int requestCode, Intent intent, FragmentActivity activity, BrowserSwitchListener listener) {
+    public void start(BrowserSwitchOptions browserSwitchOptions, FragmentActivity activity, BrowserSwitchListener listener) {
         Context appContext = activity.getApplicationContext();
+
+        Intent intent;
+        if(browserSwitchOptions.getIntent() != null) {
+            intent = browserSwitchOptions.getIntent();
+        } else {
+            intent = config.createIntentToLaunchUriInBrowser(appContext, browserSwitchOptions.getUrl());
+        }
+
+        int requestCode = browserSwitchOptions.getRequestCode();
+
         String errorMessage = assertCanPerformBrowserSwitch(requestCode, appContext, intent);
         if (errorMessage == null) {
+            JSONObject metadata = browserSwitchOptions.getMetadata();
             BrowserSwitchRequest request = new BrowserSwitchRequest(
-                    requestCode, intent.getData(), BrowserSwitchRequest.PENDING);
+                    requestCode, intent.getData(), BrowserSwitchRequest.PENDING, metadata);
             persistentStore.putActiveRequest(request, appContext);
             appContext.startActivity(intent);
         } else {
             BrowserSwitchResult result =
-                new BrowserSwitchResult(BrowserSwitchResult.STATUS_ERROR, errorMessage);
+                    new BrowserSwitchResult(BrowserSwitchResult.STATUS_ERROR, errorMessage);
```

Comment:
Extra whitespace? Or is this intentional?

Reply:
Good catch I think this was a formatting error.